### DISCUSSION TITLE
New to Webpack 5 and Encountering Issues setting up react project

### DISCUSSION THREAD

**Discussion by knightswatch3** [webpack/webpack#12864] [Q&A]
```Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| import {render} from "react-dom";
|
> render(<App/>, document.getElementById("root"));
```

I am making sure I am using the right loaders and mentioned the jsx? in my test clause for the webpack modules. But things seem to remain same. Any help please ?

**Comment by sokra** [comment]
Since you are using JSX, which is an non-standard javascript extension, you need something to convert that to javascript so that browsers understand it.

It seems like you are trying to configure a loader for that, but

> currently no loaders are configured

None of the rules apply to that file. There is something wrong with your `test` or `include`/`exclude`

**Reply by knightswatch3** [reply]
@sokra here is my webpack configuration file , could you please advise what's incorrect here ?

```module.exports = {
    mode: 'development',
    entry : {
        index: path.resolve(__dirname,"../../src/index.js")
    },
    output: {
        path: SERVE_DIR 
    },  
    devServer:{
        contentBase: './build',
        publicPath: '/',
        port: 9000
    },
    module:{
        rules: [
            {
                test: /\.jsx?$/,
                exclude: /node_modules/,
                include: APP_DIR,
                use: ['babel-loader']
            },
            {
                test: /\.(scss|css)$/,
                use: ["style-loader","css-loader","sass-loader"]
            }
        ]
    },
    devtool : 'inline-source-map' ,
    plugins: [
        new htmlWebpackPlugin({
            template: TEMPLATE
        })
    ]
}```

**Reply by knightswatch3** [reply]
Also, I tried most of the possible solutions/ways how to configure the test section for the webpack configration to be able to load the ls/jsx in the project. But nothing seems to work.

**Reply by anshumanv** [reply]
Can you also provide your babel config (better if full repo)? Make sure you have react preset for babel for it to transpile jsx

**Reply by knightswatch3** [reply]
@anshumanv here is my babel.config.json

```{
    "presets" : ["@babel/preset-env","@babel/preset-react"]
}```

**Reply by anshumanv** [reply]
Looks okay, maybe give repo link so I can debug locally?

**Reply by knightswatch3** [reply]
@anshumanv 

https://github.com/SatishChandraPalakoti/project_rockstart

**Reply by anshumanv** [reply]
Thanks,

The problem is you didn't supply your config path in `serve` command, change to this in package.json and should work - 

` "start": " webpack serve -c ./webpackConfig/webpack.dev.config.js --open Firefox",`

**Reply by knightswatch3** [reply]
@anshumanv But I am building the project before starting to serve. Am I understanding this incorrectly ?
 Because I am following this walkthrough :

https://www.valentinog.com/blog/webpack/

**Reply by anshumanv** [reply]
The bundle you create should be served by a static server, while with `webpack serve` you use `webpack-dev-server` which serves your bundled webpack assets from in-memory so it also needs to have access to webpack config since it will essentially bundle your assets using webpack and serve them in real time.

**Reply by anshumanv** [reply]
The blog you posted has a default webpack config hence it's automatically picked up by webpack without specifying in the script, meanwhile you don't have a default config so you need to specify exact path - https://webpack.js.org/api/cli/#default-configurations

**Reply by knightswatch3** [reply]
@anshumanv  
So this means, the webpack configuration file is needed at both build step and also serve step ?

**Reply by anshumanv** [reply]
yes

**Reply by knightswatch3** [reply]
And only if the config files are not in the default location ? correct ?

**Reply by anshumanv** [reply]
yes, in that case webpack will pick it up by default

**Reply by knightswatch3** [reply]
Thanks @anshumanv for helping me out. I was stuck with this for a while. And there is a related issue which I am now able to figure out what caused it.

**Reply by sokra** [reply]
> But I am building the project before starting to serve. Am I understanding this incorrectly ?

Yes. You don't need to make a dev build before serving.

**Reply by alexander-akait** [reply]
Try to remove:
```
exclude: /node_modules/,
include: APP_DIR,
```

And check you have `@babel/preset-react` in `babel` configuration

**Reply by sokra** [reply]
This is not related to the babel config. The error message says: `currently no loaders are configured to process this file`.
This means babel is not even running yet.

What's the value of `APP_DIR`? Is the processed file inside of this directory?
